tests/integration/cmd_link_unlink_test.go

```diff
@@ -108,5 +108,30 @@ var _ = Describe("odo link and unlink command tests", func() {
 			Expect(outputErr).To(ContainSubstring("been linked"))
 			helper.CmdShouldPass("odo", "unlink", "backend", "--component", "frontend", "--project", project, "--port", "8778", "--context", context2)
 		})
+
+		It("Wait till frontend dc rollout properly after linking the frontend application to the backend", func() {
+			appName := helper.RandString(7)
+			helper.CopyExample(filepath.Join("source", "nodejs"), context1)
+			helper.CmdShouldPass("odo", "create", "nodejs", "frontend", "--app", appName, "--context", context1, "--project", project)
+			helper.CmdShouldPass("odo", "url", "create", "--port", "8080", "--context", context1)
+			helper.CmdShouldPass("odo", "push", "--context", context1)
+			frontendURL := helper.DetermineRouteURL(context1)
+
+			oc.ImportJavaIS(project)
+			helper.CopyExample(filepath.Join("source", "openjdk"), context2)
+			helper.CmdShouldPass("odo", "create", "java:8", "backend", "--app", appName, "--project", project, "--context", context2)
+			helper.CmdShouldPass("odo", "url", "create", "--port", "8080", "--context", context2)
+			helper.CmdShouldPass("odo", "push", "--context", context2)
+
+			// link both component and wait till frontend dc rollout properly
+			helper.CmdShouldPass("odo", "link", "backend", "--port", "8080", "--wait", "--context", context2)
```

Comment:
`context2` is the context for the backend component. So this command is trying to link component to itself.

This should be `helper.CmdShouldPass("odo", "link", "backend", "--port", "8080", "--wait", "--context", context1)`


It would be much better if the `context1` and `context2` have some meaningful names like `backendContext` and `frontendContext` to prevent mistakes like this.

Reply:
yup, Nice catch.

Will update the context naming accordingly